.github/workflows/clean.yml

```diff
@@ -89,26 +89,14 @@ jobs:
           KUBE_ENV: ${{ inputs.app_env }}
         run: |
           source platform/lib/kube/clean.sh
+        
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      
+      - name: Checkout code
+        uses: actions/checkout@v2
 
-# TODO: Find a way to delete tag from provided docker registry and repository
-#      - name: Docker setup
-#        if: inputs.docker_username != ''
-#        shell: bash
-#        run: |
-#          #!/bin/bash
-#          set -ex
-#
-#          mkdir -p $HOME/.regctl
-#
-#          curl -L "https://github.com/regclient/regclient/releases/latest/download/regctl-linux-arm64" > "$HOME/.regctl/regctl"
-#          chmod 755 "$HOME/.regctl/regctl"
-#
-#          echo "$HOME/.regctl" >> $GITHUB_PATH
-#          regctl version
-#
-#      - name: Docker clean
-#        if: inputs.docker_username != ''
-#        shell: bash
-#        run: |
-#          regctl registry login ${{ inputs.registry }} -u ${{ inputs.username }} -p ${{ secrets.docker_password }}
-#          regctl tag rm ${{ inputs.registry }}/${{ inputs.username }}/${{ inputs.app_name }}:${{ steps.extract_branch.outputs.branch_hash }}
+      - name: Delete clean
```

Comment:
@Patel-world - This is not we want, we need a solution which works out for following cases:

- Docker registry is configurable, the repository running this workflow can be using any registry (see - `inputs.registry`). We need to make sure we run clean up for only for configured registry.
- Note that `inputs.docker_registry` is **optional**, we need to perform clean up for registry only if this input is provided. You can safely assume that if `inputs.docker_registry` is provided, `inputs.docker_username` and `secrets.docker_password` will be always present.
- Docker registry can be hosted anywhere (AWS, DigitalOcean, DockerHub). The solution you have will only work out for DockerHub. We need to make sure to continue with the clean up only if registry is a DockerHub registry (can try checking out if repository exists or not via calling the DockerHub API).
- You need to obtain auth token via configured username and password (see - `inputs.username` and `inputs.password`). **You in any case cannot check in sensitive credentials.**
- This workflow might get called multiple times for a branch. You need to make sure the delete call does not fails in case tag does not exists.